pontoon/sync/formats/ftl.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import absolute_import
+
+import codecs
+import copy
+import logging
+import os
+
+from ftl.format.parser import FTLParser
+from ftl.format.serializer import FTLSerializer
+
+from pontoon.sync import SyncError
+from pontoon.sync.formats.base import ParsedResource
+from pontoon.sync.vcs.models import VCSTranslation
+
+log = logging.getLogger(__name__)
+
+
+class L20NEntity(VCSTranslation):
+    """
+    Represents entities in l20n (without its attributes).
+    """
+    def __init__(self, key, source_string, source_string_plural, strings, comments=None, order=None):
+        super(L20NEntity, self).__init__(
+            key=key,
+            source_string=source_string,
+            source_string_plural=source_string_plural,
+            strings=strings,
+            comments=comments or [],
+            fuzzy=False,
+            order=order,
+        )
+
+    def __repr__(self):
+        return '<L20NEntity {key}>'.format(key=self.key.encode('utf-8'))
+
+
+class L20NResource(ParsedResource):
+    def __init__(self, path, locale, source_resource=None):
+        self.path = path
+        self.locale = locale
+        self.entities = {}
+        self.source_resource = source_resource
+        self.order = 0
+
+        # Copy entities from the source_resource if it's available.
+        if source_resource:
+            for key, entity in source_resource.entities.items():
+                self.entities[key] = L20NEntity(
+                    entity.key,
+                    '',
+                    '',
+                    {},
+                    copy.copy(entity.comments),
+                    entity.order
+                )
+
+        with codecs.open(path, 'r', 'utf-8') as resource:
+            self.structure = FTLParser().parseResource(resource.read())
+
+        def get_comment(obj):
+            return [obj['comment']['content']] if obj['comment'] else []
+
+        def parse_entity(obj, section_comment=[]):
+            if obj['value'] and obj['value']['type'] == 'Pattern':
```

Comment:
Ah, I see. You're just checking if the entity has a value. A value is always a Pattern